### PR TITLE
Document testing process

### PR DESCRIPTION
**Pull request by malyvsen**
It took me a while to figure it out, so now it won't for someone next :)

### CODE REVIEW

**Comment by malyvsen** [comment]
@samedii documented in main README

File: README.rst [hunk 1]
```diff
@@ -73,3 +73,13 @@ Utility functions that don't fit elsewhere.
 ~~~~~~~~~~~~~~
 
 Operations available from the `guild.ai <https://guild.ai/>`__ CLI.
+
+Testing
+-------
+
+- Before running any tests, make sure you have ``poetry`` and ``cookiecutter`` globally installed
```
Comment (samedii) [inline README.rst:80]: Why does cookiecutter have to be global? And poetry can be nonglobal in pyenv?

Comment (samedii) [reply README.rst:80]: cookiecutter can be in `poetry add --dev cookiecutter`?

File: README.rst [hunk 2]
```diff
@@ -73,3 +73,13 @@ Utility functions that don't fit elsewhere.
 ~~~~~~~~~~~~~~
 
 Operations available from the `guild.ai <https://guild.ai/>`__ CLI.
+
+Testing
```
Comment (samedii) [inline README.rst:77]: Testing and developing locally?

File: README.rst [hunk 3]
```diff
@@ -73,3 +73,13 @@ Utility functions that don't fit elsewhere.
...
+-------
+
+- Before running any tests, make sure you have ``poetry`` and ``cookiecutter`` globally installed
+- Run all scripts tests from the root directory of this repo
+- Execute the scripts rather than sourcing them
+- Run ``test/create.sh`` first to initialize a fake repo from the template - it'll end up under ``.test-template``
+- Then run all the other scripts in any order you like (except for ``test/env.sh`` - that one's just a helper)
```
Comment (samedii) [inline README.rst:84]: Easier to follow with example of step by step commands?

**Review by samedii — COMMENTED** [review]
Good start

File: README.rst (hunk 2 above)
Comment (malyvsen) [reply README.rst:77]: This is only about testing tho. You read that section if you're developing locally and then you're like "ooh, I should test this"

**Comment by malyvsen** [comment]
@samedii Should make you happier hopefully

File: pyproject.toml [hunk 4]
```diff
@@ -12,6 +12,8 @@ cookiecutter = "^1.7.2"
 [tool.poetry.dev-dependencies]
 flake8 = "^3.8.4"
 black = "^20.8b1"
+poetry = "^1.1.5"
```
Comment (samedii) [inline pyproject.toml:15]: It looks weird that poetry is a dependency since it's needed to install the deps?

**Review by samedii — APPROVED** [review]
We could add a "Development" section etc but this is a good start

**Pull request merged**